api/identitatem/v1alpha1/identityprovider_types.go

```diff
@@ -13,17 +14,40 @@ import (
 type IdentityProviderSpec struct {
```

Comment:
This list of supported identity providers looks good for now... we may want to add additional identity providers *if* we get demand.  This list should cover over 97% of use cases I'm aware of with an additional 1 - 2 % coming from Keystone.

Reply:
@robinbobbitt this is the "duck-typing" I was hoping we could do ;-)  Notice that these identity provider types are coming from the OpenShift config v1 go package.